code/modules/mob/living/living.dm

```diff
@@ -735,116 +735,66 @@ default behaviour is:
 
 	return
 
-/mob/living/Move(a, b, flag)
-	if (buckled && buckled.loc != a) //not updating position
-		if(istype(buckled, /mob))	//If you're buckled to a mob, a la slime things, keep on rolling.
-			return buckled.Move(a, b)
-		else	//Otherwise, no running around for you.
+// Almost all of this handles pulling movables behind us
+/mob/living/Move(atom/newloc, direct, movetime)
+	if(buckled && buckled.loc != newloc) //not updating position
+		if(!buckled.anchored && buckled.buckle_movable)
+			return buckled.Move(newloc, direct)
+		else
 			return 0
 
-	if (restrained())
+	var/atom/movable/pullee = pulling
+	// Prior to our move it's already too far away
+	if(pullee && get_dist(src, pullee) > 1)
+		stop_pulling()
+	// Shenanigans!
+	if(pullee && !isturf(pullee.loc) && pullee.loc != loc)
+		log_debug("[src]'s pull on [pullee] was broken despite [pullee] being in [pullee.loc]. Pull stopped manually.")
```

Comment:
This will create a debug log every time someone is pulling something and it gets shut in a locker.

Reply:
Copied from Paradise and I don't see their server on fire, but it can be removed if you think it's not desirable.

Reply:
Changed